server/routes/chefs.js

```diff
@@ -0,0 +1,119 @@
+// ROUTER FOR OUR CHEF
+const router = require("express").Router();
+
+const chefsController = require("../controllers/chefsController");
+const usersController = require("../controllers/usersController");
+const { isArrayOfStrings } = require("../util");
+
+// get a chef for an id
+router.get("/:id", async (req, res) => {
+    try {
+        const { id } = req.params;
+        if (!id) {
+            res.status(400).json({ errors: ["Missing id for chef"] });
+            return;
+        }
+        const chef = await chefsController.findOneWithId(id);
+        if (!chef) {
+            res.status(400).json({ errors: ["Chef not found for given id"] });
+            return;
+        }
+        res.json(chef);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ errors: ["Unexpected error occured"] });
+    }
+});
+
+// get the chef profile for signedin user
+router.get("/", async (req, res) => {
```

Comment:
we need to add the loginRequired middleware here

Reply:
The chefs route is protected by `loginRequired` in the `app.js` itself.

```js
app.use("/auth", authRouter);
app.get("/ping", loginRequired, (req, res) => res.json({ success: true }));
app.use("/chefs", loginRequired, chefsRouter);
app.use("/meals", loginRequired, mealsRouter);
```

The use of `loginRequired` at the `app.js` level made much more sense at the time of writing the middleware as I thought all the unprotected routes would be under `/auth` (`/auth/login` and `/auth/register`) and all the user related routes which needed middleware would be clubbed under `/users` with its own router.

We can:
1. keep it as it is
2. move all the `loginRequired` use to either root level and break `/auth` into `/auth` and `/users`
3. move all the `loginRequired` into the respective route file.

thoughts? @megangilpin

Reply:
I like that the loginRequired is in the App.js itself. At the end of the day I think it will make it more DRY and you can easily tell which routes are the protected routes. Plus it makes it easy to remove from the routes when testing on Postman. Originally, I had set it up as option 2 because I was thinking we might want to reset the token if they come back.  But now thinking about it maybe we shouldn't do that for security reasons.

Reply:
Sorry, we didn't bring this to a conclusion during the stand-up today. Let's take a vote on what we should do.

I prefer `2.`

@sagar-13 @sina-jamshidi please vote!

Reply:
Did some reading and I think I like option of keeping it as it is.(just to make it less repetitive and also in the interest of time).

Reply:
As per the voting, I have split the auth router into auth and users routes